src/App.tsx

```diff
@@ -15,48 +17,126 @@ export const goodsFromServer = [
   'Garlic',
 ];
 
+enum SortingParams {
+  Name = 'name',
+  Length = 'length',
+  Default = '',
+}
+
+interface SortOptions {
+  option: SortingParams,
+  reversed: boolean,
+}
+
+const getVisibleGoods = (
+  goods: typeof goodsFromServer,
+  { option, reversed }: SortOptions,
+) => {
+  let visibleGoods = goods;
+
+  if (option) {
+    visibleGoods = [...visibleGoods].sort((good1, good2) => {
+      switch (option) {
+        case SortingParams.Name:
+          return good1.localeCompare(good2);
+        case SortingParams.Length:
+          return good1.length - good2.length;
+        default:
+          return 0;
+      }
+    });
+  }
+
+  if (reversed) {
+    visibleGoods = [...visibleGoods].reverse();
+  }
+
+  return visibleGoods;
+};
+
 export const App: React.FC = () => {
+  const [sortOptions, setSortOptions] = useState<SortOptions>({
+    option: SortingParams.Default,
+    reversed: false,
+  });
+
+  const visibleGoods = getVisibleGoods(
+    goodsFromServer,
+    sortOptions,
+  );
+
   return (
     <div className="section content">
       <div className="buttons">
         <button
           type="button"
-          className="button is-info is-light"
+          className={cn(
+            'button',
+            'is-info',
+            { 'is-light': sortOptions.option !== SortingParams.Name },
+          )}
+          onClick={() => setSortOptions({
+            ...sortOptions,
+            option: SortingParams.Name,
+          })}
         >
           Sort alphabetically
         </button>
 
         <button
           type="button"
-          className="button is-success is-light"
+          className={cn(
+            'button',
+            'is-success',
+            { 'is-light': sortOptions.option !== SortingParams.Length },
+          )}
+          onClick={() => setSortOptions({
+            ...sortOptions,
+            option: SortingParams.Length,
+          })}
         >
           Sort by length
         </button>
 
         <button
           type="button"
-          className="button is-warning is-light"
+          className={cn(
+            'button',
+            'is-warning',
+            { 'is-light': !sortOptions.reversed },
+          )}
+          onClick={() => setSortOptions({
+            ...sortOptions,
+            reversed: !sortOptions.reversed,
+          })}
         >
           Reverse
         </button>
 
-        <button
-          type="button"
-          className="button is-danger is-light"
-        >
-          Reset
-        </button>
+        {visibleGoods !== goodsFromServer
```

Comment:
this condition is wrong because two arrays can't be the same in js. The same can be only links of the arrays.
for example: try to console this expression:
`console.log([] === [])` or 'console.log([ { id: 1 } ] === [ { id: 1 } ])'

Reply:
Of course I understand it, in my logic I just want to check if visibleGoods variable consists link to original goodsFromServer. Because I return from function either changed copy of original goodsFromServer array or just link to it, that will mean that there aren't any changes comparing to original goods and reset button shouldn't be visible

Reply:
I just want to now what approach to conditional rendering in this case will be better if I'll change my function according to your suggestion. Btw, my code also worked.

Reply:
I changed my code, but now it seems to me that conditional rendering became less elegant